lib/spack/spack/package.py

```diff
@@ -1420,6 +1524,43 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class Package(PackageBase):
+    phases = ['install', 'log']
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
+
+
+class AutotoolsPackage(PackageBase):
+    phases = ['autoreconf', 'configure', 'build', 'install', 'log']
+
+    def autoreconf(self, spec, prefix):
+        """Not needed usually, configure should be already there"""
+        pass
+
+    @PackageBase.sanity_check('autoreconf')
+    def is_configure_or_die(self):
+        if not os.path.exists('configure'):
+            raise RuntimeError('configure script not found in {0}'.format(os.getcwd()))
+
+    def configure_args(self):
+        return list()
```

Comment:
@alalazo I agree that as someone familiar with Python, deriving everything from self helps me understand how everything is organized. But a lot of package developers don't know Python and shouldn't have to. So far, writing your own package is really simple. As soon as we start requiring people to understand objects, we limit the number of people who can create basic packages.

By the way, another sanity check that NEEDS to be in place for PythonPackage is simply to try importing the package. I've run into SO MANY cases where I installed a package just fine, but it crashes when I try to import it.